pkg/controller/atlasdatabaseuser/databaseuser.go

```diff
@@ -180,6 +214,21 @@ func userMatchesSpec(log *zap.SugaredLogger, atlasSpec *mongodbatlas.DatabaseUse
 		return false, err
 	}
 
+	// performing some normalization of dates
```

Comment:
this is necessary to correctly compare strings as the date returned by Atlas may be different than the one submitted by the Atlas Operator